src/index.js

```diff
@@ -46,27 +94,88 @@ const handleItemMenuClick = () => {
       input.focus();
       parent.classList.add('active');
 
-      deleteIcon.addEventListener('click', () => {
-        parent.parentElement.removeChild(parent);
+      input.addEventListener('change', (e) => {
+        e.preventDefault();
+        let fromLocal = JSON.parse(window.localStorage.getItem('todo-tasks'));
+
+        fromLocal = fromLocal.map((task) => {
+          if (task.id === checkBoxId) {
+            const obj = {
+              ...task,
+              description: input.value,
+              completed: false,
+            };
+            return obj;
+          }
+          return task;
+        });
+        // we can as well rerender the updated item to the ui
+        document.querySelector('.todo-list').innerHTML = '';
+        populateList(fromLocal);
+        handleOnCheckboxClick();
+        handleItemMenuClick();
+
+        window.localStorage.setItem('todo-tasks', JSON.stringify(fromLocal));
```

Comment:
- [ ] [Optional] It'll be nice if after adding a new task, the input field is cleared.